src/impy/cli.py

```diff
@@ -0,0 +1,348 @@
+    return Particle.from_pdgid(pid).programmatic_name
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-v", "--version", action="store_true", help="print version")
+    parser.add_argument(
+        "-s",
+        "--seed",
+        type=int,
+        default=0,
+        help="seed between 1 and 1e9 (default is 0, which generates random seed)",
+    )
+    parser.add_argument(
+        "-n",
+        "--number",
+        type=int,
+        default=1,
+        help="number of collisions (default is 1)",
+    )
+    parser.add_argument(
+        "-m",
+        "--model",
+        default=0,
+        help=f"select model via number or name with tolerant matching {VALID_MODELS}",
+    )
+    parser.add_argument(
+        "-p",
+        "--projectile-momentum",
+        type=float,
+        default=0,
+        help="projectile momentum in GeV/c",
+    )
+    parser.add_argument(
+        "-P",
+        "--target-momentum",
+        type=float,
+        default=0,
+        help="target momentum in GeV/c",
+    )
+    parser.add_argument("-S", "--sqrts", type=float, default=0, help="sqrt(s) in GeV")
+    parser.add_argument(
+        "-i",
+        "--projectile-id",
+        default=2212,
+        help="PDG ID or particle name, default is proton",
+    )
+    parser.add_argument(
+        "-I",
+        "--target-id",
+        default=2212,
+        help="PDG ID or particle name, default is proton",
+    )
+    parser.add_argument(
+        "-o", "--output", default="", help=f"output format {VALID_FORMATS}"
+    )
+    parser.add_argument(
+        "-f", "--out", help="Output file name (generated if none provided)"
+    )
+
+    args = parser.parse_args()
+
+    if args.version:
+        print(f"impy {version}")
+        raise SystemExit
+
+    max_seed = int(1e9)  # EPOS requirement
+    if args.seed <= 0:
+        args.seed = int.from_bytes(os.urandom(4), "little")
+        args.seed %= max_seed  # result of modulus is always positive
+    elif args.seed > max_seed:
+        raise SystemExit(
+            f"Error: {args.seed} is larger " f"than maximum seed ({max_seed})"
+        )
+
+    if args.number <= 0:
+        raise SystemExit("Error: number must be positive")
+
+    try:
+        model_number = int(args.model)
+        Model = MODELS[model_number]
+    except KeyError:
+        raise SystemExit(f"Error: model number {args.model} is invalid {VALID_MODELS}")
+    except ValueError:
+        # args.model is not a number.
+        # Find model that matches spec.
+        matches = []
+        x = args.model.lower()
+        for M in get_all_models():
+            y = M.label.lower()
+            # If exact match found, stop search. This rule is necessary to e.g. select
+            # Sibyll-2.3, since Sibyll-2.3d would also match otherwise, always leading to
+            # ambiguity.
+            if x == y:
+                Model = M
+                break
+            if tolerant_string_match(x, y):
```

Comment:
I added some comments and you can also look into the unit test to see some examples `test_model...` in `test_cli.py`.

For several good reasons, we should specify models not by number (not human-friendly) but by name. We could just ask the user to pass the exact model label as a string (what Model.label returns), but that's not very user friendly. This code here
- Ignores case, so one does not have to remember typing SIBYLL or Sibyll.
- Allows you to write just "epos" to match "EPOS-LHC", because that's the only version of EPOS we have. This is done by requiring that all characters in the spec are in the model label and in correct order. See the unit test for `tolerant_string_match` in `test_util.py` for a few examples.
- The spec can be ambiguous, for example, when you request `sibyll`, in that case the code tells you what matches are found, this allows you to refine your spec.
- As a special case, if your spec matches a model exactly, you will get that model even if there is another model that would match `tolerant_string_match`, too. Without that rule, you could not select "Sibyll-2.3" because it is a substring of "Sibyll-2.3d"